### ISSUE TITLE
Sub seconds on Rule.run_in()?

### ISSUE THREAD

**Issue by goodfore** [spacemanspiff2007/HABApp#71]
Would it be possible to add support for sub second delay for methods like run_in()?

Thanks,
Joe

**Comment by spacemanspiff2007** [comment]
Currently the scheduler only runs every second to keep performance impact light.
Maybe if I start refactoring I'll add the possibility to add sub second calls.
Meanwhile just use ````time.sleep```` to sleep the desired subsecond part.

**Comment by goodfore** [comment]
Thanks. That's what I am using currently.

**Comment by NickWaterton** [comment]
I'm running into the same thing, could we use
```
await asyncio.sleep(0.1)
```
To avoid messing up the scheduler - would not a blocking call (to time.sleep0 ) suspend all rules execution?

**Comment by spacemanspiff2007** [comment]
Yes sure, this works fine. But then you have to use an ``async`` function and make sure that there are no other blocking calls in that function. A workaround could be
```python
async def wait_time(self, event):
    await asyncio.sleep(0.1)
    self.run_soon(self.my_blocking_func)
```

**Comment by spacemanspiff2007** [comment]
Fixed with 0.30.0
